libs/localization/scripts/extract.ts

```diff
@@ -41,105 +40,91 @@ spawn.sync('npx', [
   process.argv[2],
 ])
 
-function createNamespace(
-  id: string,
-  messages: MessageDict,
-  locales: Locale[],
-  space: string,
-) {
-  const emptyObjForEachLocale = locales.reduce(
-    (arr, curr) => ({
-      ...arr,
-      [curr.code]: {},
-    }),
-    {},
-  )
+const getNamespace = (id: string, space: string) =>
+  client
+    .getSpace(space)
+    .then((space) => space.getEnvironment('master'))
+    .then((environment) => environment.getEntry(id))
+    .catch(() => null)
 
-  return client
+const createNamespace = (id: string, messages: MessageDict, space: string) =>
+  client
     .getSpace(space)
     .then((space) => space.getEnvironment('master'))
     .then((environment) =>
-      environment.createEntryWithId('namespace', id, {
+      environment.createEntryWithId('namespaceJeremyDev', id, {
+        // TODO: put namespace back after review
         fields: {
           namespace: {
             'is-IS': id,
           },
-          defaults: {
-            'is-IS': messages,
+          strings: {
+            'is-IS': convertToLocalArray(messages),
           },
-          fallback: emptyObjForEachLocale,
-          strings: emptyObjForEachLocale,
         },
       }),
     )
     .catch((err) => console.log(err))
-}
-
-function updateNamespace(namespace: Entry, messages: MessageDict) {
-  namespace.fields.defaults['is-IS'] = Object.assign(
-    {},
-    namespace.fields.defaults['is-IS'],
-    messages,
-  )
 
-  const newStrings = Object.keys(messages).reduce(
-    (arr, cur) => ({ ...arr, [cur]: '' }),
-    {},
-  )
+export const mergeArray = (local: DictArray[], distant: DictArray[]) =>
+  local.map((localObj) => {
+    const contentfulValue = distant.find(
+      (contentfulObj) => contentfulObj.id === localObj.id,
+    )
+    const hasIs = !isEmpty(contentfulValue?.['is-IS'])
+    const hasEn = !isEmpty(contentfulValue?.en)
+
+    return {
+      ...localObj,
+      'is-IS': hasIs ? contentfulValue!['is-IS'] : localObj['is-IS'],
+      en: hasEn ? contentfulValue!.en : localObj.en,
+    }
+  })
 
-  namespace.fields.strings = Object.keys(namespace.fields.strings).reduce(
-    (arr, cur) => ({
-      ...arr,
-      [cur]: Object.assign({}, newStrings, namespace.fields.strings[cur]),
-    }),
-    {},
-  )
+export const convertToLocalArray = (messages: MessageDict) =>
+  Object.keys(messages).map((item) => ({
+    id: item,
+    defaultMessage: messages[item].defaultMessage,
+    'is-IS': messages[item].defaultMessage,
+    en: messages[item].description,
+  }))
+
+export const convertToContentfulArray = (namespace: Entry) =>
+  (namespace?.fields?.strings?.['is-IS'] ?? []).map((item: DictArray) => ({
+    id: item.id,
+    defaultMessage: item.defaultMessage,
+    'is-IS': item['is-IS'],
+    en: item.en,
+  }))
+
+export const updateNamespace = (namespace: Entry, messages: MessageDict) => {
+  const fromLocal = convertToLocalArray(messages)
+  const fromContentful = convertToContentfulArray(namespace)
```

Comment:
Regarding the function names, maybe Objective C style `translationsFromLocal` and `translationsFromContentful` is clearer?